packages/venia-concept/src/components/Navigation/navigation.js

```diff
@@ -236,6 +291,7 @@ class Navigation extends PureComponent {
                 <div className={classes.footer}>{footer}</div>
                 {signInForm}
                 {createAccountForm}
+                {forgotPasswordForm}
```

Comment:
Okay I think this answers the question:

> Why are we waiting for SignIn's onForgotPasswordto set this.forgotPassword?

Before `onForgotPassword` is clicked, this always returns `null` and won't render anything.

But unless I'm missing something this doesn't change anything visually in the app, so if we can reduce the complexity there, we should.

Reply:
Right now our input component handlers initial values only when component is mounted